### PR TITLE
Add puzzle dataset generation script

### PR DESCRIPTION
**Pull request by adeelahmad**
## Summary
- add `puzzle_dataset_generation.py` for generating synthetic puzzle datasets with MCQ, Q&A, structured extraction, and ambiguity handling
- include CLI supporting generation, validation, and sampling
- provide pytest suite covering generation, schema validation, and CLI sample output
- update trainer to load tokenizers using `load_tokenizer` to avoid redundant model loads
- remove temporary `llm_templates` helper and inline fallback classes so the generator is self-contained

## Testing
- `pytest -q`
- `python puzzle_dataset_generation.py sample --n 1 --mode mcq --seed 0`

------
https://chatgpt.com/codex/tasks/task_e_68bbdf658624832eadd58bb109c24289

### CODE REVIEW

File: mlx_grpo_trainer_aligned.py [hunk 1]
```diff
@@ -7191,11 +7191,12 @@ def main():
...
-    ref_model, ref_tokenizer = load(Path(args.ref_model_path))  # CONTINUE HERE
+    ref_model, _ = load(Path(args.ref_model_path))
     ref_model.freeze()
     ref_model.eval()
 
     # Reference tokenizer (teacher)
+    ref_tokenizer = getattr(args, "_ref_tokenizer", None)
     setattr(args, "_ref_tokenizer", ref_tokenizer)
 
     # Build optional alignment bridge
```
[the remark this answers is not in the document]
Comment (adeelahmad) [reply mlx_grpo_trainer_aligned.py:7202]: @codex fix it.
